### PR TITLE
Add task reporting after every async action in the strapi logs

### LINKED ISSUE #537: feedback status when tasks are running
**Description**
When we try to update or make the first sync on big amount of data a lot of tasks can be in queued.
Because no feedback are seen, we may check the box many time and juste crash the cloud or make sync append many times.

It should be great to add a feedback about how many indexing operations left and add a warning if too many of them are left before clicking on update

**Basic example**
it can also be usefull when  we use the same meilisearch instance for many indexes

### PR DESCRIPTION
**Pull request by bidoubiwa**
Maybe fixes #537 

The task numbers for every async operation are now communicated in the strapi logs.

### CODE REVIEW

File: server/services/meilisearch/connector.js [hunk 1]
```diff
@@ -116,12 +122,18 @@ module.exports = ({ strapi, adapter, config }) => {
...
-          return client.index(indexUid).deleteDocument(
+          const task = client.index(indexUid).deleteDocument(
             adapter.addCollectionNamePrefixToId({
               contentType,
               entryId: entry.id,
             })
           )
+
+          strapi.log.info(
+            `Create task to delete 1 document of the index "${indexUid}" in Meilisearch (Task uid: ${task.taskUid})`
```
Comment (bidoubiwa) [inline server/services/meilisearch/connector.js:133]:
```suggestion
            `A task to delete one document from the Meilisearch index "${indexUid}" has been enqueued (Task uid: ${task.taskUid}).`
```

File: server/services/meilisearch/connector.js [hunk 2]
```diff
@@ -245,6 +257,9 @@ module.exports = ({ strapi, adapter, config }) => {
       })
 
       const task = await client.index(indexUid).addDocuments(documents)
+      strapi.log.info(
+        `Create task to add ${documents.length} documents in the index "${indexUid}" to Meilisearch (Task uid: ${task.taskUid})`
```
Comment (bidoubiwa) [inline server/services/meilisearch/connector.js:261]:
```suggestion
        `The task to add ${documents.length} documents to the Meilisearch index "${indexUid}" has been enqueued (Task uid: ${task.taskUid}).`
```

File: server/__tests__/meilisearch.test.js [hunk 3]
```diff
@@ -145,13 +145,15 @@ describe('Tests content types', () => {
...
       },
     })
 
     await service.addContentTypeInMeiliSearch({ contentType: 'restaurant' })
 
-    expect(Meilisearch().index).toHaveBeenCalledWith('restaurant')
-    expect(Meilisearch().index().addDocuments).toHaveBeenNthCalledWith(
+    const client = new Meilisearch({ host: '' })
+    expect(client.index).toHaveBeenCalledWith('restaurant')
+    expect(client.index('restaurant').addDocuments).toHaveBeenNthCalledWith(
```
Comment (brunoocasali) [inline server/__tests__/meilisearch.test.js:156]: Why did you have to change these lines?

File: server/services/meilisearch/connector.js [hunk 4]
```diff
@@ -99,7 +99,13 @@ module.exports = ({ strapi, adapter, config }) => {
         adapter.addCollectionNamePrefixToId({ entryId, contentType })
       )
 
-      return await client.index(indexUid).deleteDocuments(documentsIds)
+      const task = await client.index(indexUid).deleteDocuments(documentsIds)
+
+      strapi.log.info(
```
Comment (brunoocasali) [inline server/services/meilisearch/connector.js:104]: Can we have a test to cover those calls?

File: server/__tests__/meilisearch.test.js (hunk 3 above)
Comment (bidoubiwa) [reply server/__tests__/meilisearch.test.js:156]: My typescript plugin on my IDE was not happy about it 😅

File: server/services/meilisearch/connector.js (hunk 4 above)
Comment (bidoubiwa) [reply server/services/meilisearch/connector.js:104]: I covered the ones that could be simply tested. The other ones are harder. Related to this issue https://github.com/meilisearch/strapi-plugin-meilisearch/issues/64

**Review by brunoocasali — APPROVED** [review]
Seems good to me

bors merge

**Pull request merged**